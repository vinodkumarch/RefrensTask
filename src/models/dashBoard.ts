export interface DashBoard {
  gender: string;
  image: string;
  name: string;
  species: string;
  status: string;
}
