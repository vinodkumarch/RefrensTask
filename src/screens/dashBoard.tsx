import * as React from 'react';
import {ActivityIndicator, FlatList} from 'react-native';
import {RenderItem} from '../components/renderItem';
import {dashBoardStyles} from './dashBoard.styles';
import {useDashBoard} from './useDashBoard';

export const DashBoard = () => {
  const {gridData, loading, onUserProfilePress} = useDashBoard();
  return loading ? (
    <ActivityIndicator
      size={'large'}
      style={dashBoardStyles.loadingIndicator}
    />
  ) : (
    <FlatList
      testID="dashBoard.mainContainer"
      style={dashBoardStyles.container}
      data={gridData}
      renderItem={({item, index}) => (
        <RenderItem
          item={item}
          onUserProfilePress={() => onUserProfilePress(index)}
        />
      )}
      numColumns={2}
    />
  );
};
