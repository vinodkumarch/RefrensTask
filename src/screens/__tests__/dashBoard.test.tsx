import React from 'react';
import {DashBoard} from '../dashBoard';
import {render, waitFor} from '@testing-library/react-native';

describe('UT for DashBoard screen', () => {
  it('should reder the dashboard view', async () => {
    const {toJSON, getByTestId} = render(<DashBoard />);
    await waitFor(() =>
      expect(getByTestId('dashBoard.mainContainer')).toBeTruthy(),
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
