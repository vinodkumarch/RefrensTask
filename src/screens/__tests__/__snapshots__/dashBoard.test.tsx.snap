// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UT for DashBoard screen should reder the dashboard view 1`] = `
<RCTScrollView
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "flexGrow": 1,
      "marginTop": 20,
    }
  }
  testID="dashBoard.mainContainer"
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
